IdentityCore/src/requests/MSIDSilentTokenRequest.m

```diff
@@ -102,6 +103,22 @@ - (void)executeRequestWithCompletion:(MSIDRequestCompletionBlock)completionBlock
 
 - (void)executeRequestImpl:(MSIDRequestCompletionBlock)completionBlock
 {
+    // We return early if account metadata shows account in signed-out state
+    NSError *signInStateError;
```

Comment:
I'm thinking if this logic should be on MSAL and broker level?
Two things to consider:
1. Broker has silent lookups outside of silent token request (e.g. usage of PRT) that need to be gated on account metadata, might be worth doing it one time.
2. If we ever use this class in ADAL, we'll do an unneeded lookup into cache.

Reply:
changed ;)